app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecFragment.kt

```diff
@@ -47,41 +70,145 @@ class RecFragment(contentLayoutId: Int) : Fragment(contentLayoutId) {
+            )
+            Spacer(Modifier.size(ButtonDefaults.IconSpacing))
+            Text(text.value)
+        }
+    }
+
+    private fun onTickTimer(interval: Long, onTick: (Long) -> Unit) : CountDownTimer {
+        return object : CountDownTimer(MILLIS_IN_FUTURE, interval) {
+            override fun onTick(millisUntilFinished: Long) {
+                onTick(millisUntilFinished)
+            }
+
+            override fun onFinish() {
+                // Do nothing
             }
         }
     }
 
+    private fun switchMetronomeState(metronomeButtonText: MutableState<String>, tempoValue: MutableState<String>) {
+        if (metronomeIsPlaying) {
+            metronomeButtonText.value = "Start metronome"
+            metronomeIsPlaying = false
+            metronomeTimer.cancel()
+        } else {
+            val tempo : Long = try {
+               tempoValue.value.toLong()
+            } catch (_: java.lang.NumberFormatException) {
+                tempoValue.value = "60"
+                60L
+            }
+
+            metronomeTimer = onTickTimer(1000 * 60 / tempo) {
+                if (this::mediaPlayer.isInitialized) {
+                    if (mediaPlayer.isPlaying) {
+                        mediaPlayer.pause()
+                        mediaPlayer.seekTo(0)
+                    }
+                    mediaPlayer.start()
+                }
+            }
+
+            metronomeTimer.start()
+            metronomeIsPlaying = true
+            metronomeButtonText.value = "Stop metronome"
+        }
+    }
+
+    private fun switchRecordState(counterText : MutableState<String>, recordButtonText : MutableState<String>) {
+        if (!isRecording) {
+            // Set the timer
+            recordTimer = onTickTimer(COUNT_DOWN_INTERVAL) { millisUntilFinished ->
+                counterText.value = ((MILLIS_IN_FUTURE - millisUntilFinished) / Companion.COUNT_DOWN_INTERVAL).toString()
+            }
+
+            //start recording
+            recordButtonText.value = "Stop recording"
+            // Start the timer
+            recordTimer.start()
+            startRecording()
+        } else {
+            //stop recording
+            // Stop the timer
+            recordTimer.cancel()
+            // Set the recording time to 0
+            counterText.value = "Recording saved!"
+            recordButtonText.value = "Start recording"
+            stopRecording()
+        }
+    }
+
+
     fun getOutputFilePath(): String {
         return requireContext().externalCacheDir?.absolutePath + "/recording.3gp"
     }
 
     @SuppressLint("Permissions are checked before calling this method", "MissingPermission")
     private fun startRecording() {
         // Initialize the AudioRecord
-        audioRecorder = AudioRecord(audioSource, sampleRateInHz, channelConfig, audioFormat, bufferSize)
+        audioRecorder = AudioRecord(audioSource,
+            Companion.SAMPLE_RATE_IN_HZ, channelConfig, audioFormat, bufferSize)
```

Comment:
I have a question: what is this Companion used to find the Sample rate?

Reply:
It is from the companion object of the class. Companion objects are used to store consts or other class members that could be used outside of the class or independently of an instance of a class. Here I used that only to separate constances from the others parts of the class. 
I could also remove the `Companion.<...>`part as it can be implicit